server/service/src/rnr_form/schedules_with_periods.rs

```diff
@@ -0,0 +1,63 @@
+use crate::service_provider::ServiceContext;
+
+use chrono::Utc;
+use repository::{
+    DateFilter, EqualFilter, Period, PeriodFilter, PeriodRepository, PeriodScheduleRow,
+    PeriodScheduleRowRepository, PeriodSort, PeriodSortField,
+    ProgramRequisitionSettingsRowRepository, RepositoryError,
+};
+
+#[derive(Debug)]
+pub struct PeriodSchedule {
+    pub schedule_row: PeriodScheduleRow,
+    pub periods: Vec<Period>,
+}
+
+pub fn get_schedules_with_periods_by_program(
+    ctx: &ServiceContext,
+    program_id: &str,
+) -> Result<Vec<PeriodSchedule>, RepositoryError> {
+    let settings_repo = ProgramRequisitionSettingsRowRepository::new(&ctx.connection);
+    let period_repo = PeriodRepository::new(&ctx.connection);
+
+    let settings = settings_repo.find_many_by_program_id(program_id)?;
+
+    let mut period_schedule_ids = settings
+        .iter()
+        .map(|s| s.period_schedule_id.clone())
+        .collect::<Vec<String>>();
+
+    // There can be duplicates in the program settings due to name tags
```

Comment:
Good to have the comment here.
Could also consider deduping in the SQL layer. Would probably require a dedicated repository method but I'm ok with those...